docs/JestObjectAPI.md

```diff
@@ -669,7 +669,7 @@ By default, `jest.spyOn` also calls the **spied** method. This is different beha
 
 :::tip
 
-Since `jest.spyOn` is a mock, you could restore the initial state calling [jest.restoreAllMocks](#jestrestoreallmocks) on [afterEach](GlobalAPI.md#aftereachfn-timeout) method.
+Since `jest.spyOn` is a mock, you could restore the initial state by passing [jest.restoreAllMocks](#jestrestoreallmocks) to the [afterEach](GlobalAPI.md#aftereachfn-timeout) method.
```

Comment:
Hm.. I think `jest.restoreAllMocks` is actually called. Here docs speak about this pattern:

```ts
afterEach(() => {
  jest.restoreAllMocks();
});
```

A callback is *passed* to `afterEach` hook, but `jest.restoreAllMocks` is *called* inside the body of that callback. How to describe this in less clumsy way?

Reply:
We have two options here:

1. Instead of creating a callback that only contains `jest.restoreAllMocks()`, one could also directly pass `jest.restoreAllMocks` as the callback:
   ``` javascript
   afterEach(jest.restoreAllMocks);
   ```
   With this option, I think the wording I have proposed in the PR is adequate. Since this is perhaps a less conventional way of doing things, it might be worth adding this code example to the tip in the docs. Thoughts?

2. If we go with your pattern, I propose to use the following wording:
   
   > Since `jest.spyOn` is a mock, you could restore the initial state by including [`jest.restoreAllMocks()`](#jestrestoreallmocks) in the body of the callback passed to the [`afterEach`](GlobalAPI.md#aftereachfn-timeout) method.
   

What do you think?

Reply:
The shortcut might be working, but it gives a type error:

<img width="720" alt="Screenshot 2023-01-14 at 16 48 07" src="https://user-images.githubusercontent.com/72159681/212477792-64e0ad61-7aac-404d-8421-83cbf3d5f75e.png">

If I remember it correctly, an issue was opened recently regarding this type error, but it was concluded that this pattern should not be promoted.

I like the second option you proposed. Just wondering, why do you avoid saying: "by calling `jest.restoreAllMocks()`"?

Reply:
I see, thanks for this information!

Thank you for the improved phrasing of the second version. I'll make a commit for this:
> Since `jest.spyOn` is a mock, you could restore the initial state by calling [`jest.restoreAllMocks`](#jestrestoreallmocks) in the body of the callback passed to the [`afterEach`](GlobalAPI.md#aftereachfn-timeout) method.

Reply:
Thanks. By the way, how does "`afterEach` hook" sound for you? In [Setup and Teardown](https://jestjs.io/docs/setup-teardown) the `before*` and `after*` APIs are called hooks. Perhaps that could work here too?

Reply:
Are you proposing the following? That sounds good to me.
> Since `jest.spyOn` is a mock, you could restore the initial state by calling [`jest.restoreAllMocks`](#jestrestoreallmocks) in the body of the callback passed to the [`afterEach`](GlobalAPI.md#aftereachfn-timeout) hook.

Reply:
Great! And thanks for caching those missing "the" here and there. Some of them were my mistakes. Good to see them being fixed.

Reply:
👍